src/test/java/com/jcabi/s3/AwsBucketTest.java

```diff
@@ -106,4 +115,121 @@ public void existsThrowsIOException() throws IOException {
+
+    /**
+     * Returns a mocked region for supportsListingLargeBuckets test.
+     * @param data Test data
+     * @return Mocked region
+     */
+    private Region mockRegionForListingLargeBuckets(
+        final ListingLargeBucketsTestData data
+    ) {
+        final Region region = Mockito.mock(Region.class);
+        final AmazonS3 aws = Mockito.mock(AmazonS3.class);
+        Mockito.when(region.aws()).thenReturn(aws);
+        final ObjectListing firstResponse = Mockito.mock(ObjectListing.class);
+        Mockito.when(firstResponse.getNextMarker()).thenReturn(
+            data.secondItem()
+        );
+        final List<S3ObjectSummary> firstSummaries =
+            new ArrayList<S3ObjectSummary>(1);
+        final S3ObjectSummary firstSummary = new S3ObjectSummary();
+        firstSummary.setKey(data.firstItem());
+        firstSummaries.add(firstSummary);
+        Mockito.when(firstResponse.getObjectSummaries())
+            .thenReturn(firstSummaries);
+        final ObjectListing secondResponse = Mockito.mock(ObjectListing.class);
+        Mockito.when(secondResponse.getNextMarker()).thenReturn(null);
+        final List<S3ObjectSummary> secondSummaries =
+            new ArrayList<S3ObjectSummary>(1);
+        final S3ObjectSummary secondSummary = new S3ObjectSummary();
+        secondSummary.setKey(data.secondItem());
+        secondSummaries.add(secondSummary);
+        Mockito.when(secondResponse.getObjectSummaries())
+            .thenReturn(secondSummaries);
+        Mockito.when(aws.listObjects(Mockito.any(ListObjectsRequest.class)))
+            .thenAnswer(
+                //@checkstyle AnonInnerLengthCheck (25 lines)
+                new Answer<ObjectListing>() {
+                    private int counter;
+                    @Override
+                    public ObjectListing answer(
+                        final InvocationOnMock invocation) {
+                        final ListObjectsRequest request =
+                            (ListObjectsRequest) invocation.getArguments()[0];
+                        Assert.assertEquals(
+                            data.bucketName(), request.getBucketName()
+                        );
+                        Assert.assertEquals(data.prefix(), request.getPrefix());
+                        final ObjectListing result;
+                        if (counter == 0) {
+                            Assert.assertNull(request.getMarker());
+                            result = firstResponse;
+                        } else {
+                            Assert.assertEquals(
+                                data.secondItem(), request.getMarker()
+                            );
+                            result = secondResponse;
+                        }
+                        counter = counter + 1;
+                        return result;
+                    }
+                }
+            );
+        return region;
+    }
+
+    private static class ListingLargeBucketsTestData {
```

Comment:
do we need this class? Maybe it's better to hardcode all these values right in the test?